src/HigherLower/Game.hs

```diff
@@ -10,8 +10,8 @@ data Game = Game
             , status :: Status
             } deriving (Show)
 
-guess :: (Monad m) => Int -> StateT Game m (Maybe Ordering)
-guess i = state step
+checkGuess :: (Monad m) => Int -> StateT Game m (Maybe Ordering)
+checkGuess i = state step
```

Comment:
This function isn't what is letting the user guess, it's actually only verifying your guess `i` agaisn't the game's state secret so I renamed it accordingly.